Data/AnswerComparer.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Data
+{
+    class AnswerComparer : IEqualityComparer<Answer>
+    {
+        public bool Equals(Answer a1, Answer a2)
+        {
+            if (a1 == null && a2 == null)
+                return true;
+            else
+            if (a1 == null || a2 == null)
+                return false;
+            else
+            if (a1.AnswerText == a2.AnswerText)
+                return true;
+            else
+                return false;

```

Comment:
You can simplify it to
return a1.AnswerText == a2.AnswerText
with all expressions